docs/rl-algorithms/rpo.md

```diff
@@ -0,0 +1,190 @@
+    
+    === "MuJoCo v2"
+
+        |                     | ppo_continuous_action_8M ({'tag': ['v1.0.0-13-gcbd83f6']})   | rpo_continuous_action ({'tag': ['pr-331']})   |
+        |:--------------------|:-------------------------------------------------------------|:----------------------------------------------|
+        | HumanoidStandup-v2  | 108982.78 ± 17996.44                                         | 156755.15 ± 10672.52                          |
+        | Humanoid-v2         | 586.11 ± 44.21                                               | 723.01 ± 103.07                               |
+        | InvertedPendulum-v2 | 861.44 ± 27.30                                               | 857.88 ± 30.67                                |
+        | Walker2d-v2         | 3219.50 ± 948.84                                             | 3934.37 ± 428.82                              |
+
+
+        Learning curves:
+        ![](../rpo/mujoco_v2_part1.png)
+
+        The following environments require tuning of `alpha` (Algorithm 1, line 13, paper: https://arxiv.org/pdf/2212.07536.pdf). As described in the paper, this variable should be tuned for environments tested. A larger value means more randomness, whereas a smaller value indicates less randomness. Some mujoco environments require a smaller `alpha=0.01` value to achieve a reasonable performance compared to `alpha=0.5` for the rest of the environments. This version (`alpha=0.01`) of runs is indicated as `rpo_continuous_action_alpha_0_01` in the table and learning curves.
+
+
+        |                           | ppo_continuous_action_8M ({'tag': ['v1.0.0-13-gcbd83f6']})   | rpo_continuous_action_alpha_0_01 ({'tag': ['pr-331']})   |
+        |:--------------------------|:-------------------------------------------------------------|:---------------------------------------------------------|
+        | Ant-v2                    | 2435.03 ± 955.57                                             | 3142.98 ± 787.87                                         |
+        | HalfCheetah-v2            | 2714.14 ± 1260.34                                            | 2716.46 ± 1241.38                                        |
+        | Hopper-v2                 | 2411.13 ± 657.94                                             | 2286.05 ± 590.10                                         |
+        | InvertedDoublePendulum-v2 | 5727.86 ± 367.86                                             | 5811.94 ± 281.47                                         |
+        | Reacher-v2                | -4.57 ± 0.48                                                 | -4.17 ± 0.22                                             |
+        | Swimmer-v2                | 132.43 ± 10.01                                               | 141.49 ± 8.47                                            |
+        | Pusher-v2                 | -33.60 ± 8.54                                                | -26.08 ± 2.32                                            |
```

Comment:
This is interesting. Do you have the results for `alpha=0.5` for these environments? Do they make the performance a lot worse?

Reply:
Yes, the performance is generally worse for `alpha=0.5` for those environments. They are very sensitive to random perturbation on the mean. Here are the results for Mujoco v4 (v2 results are similar).

|                           | ppo_continuous_action_8M ({'tag': ['v1.0.0-13-gcbd83f6']})   | rpo_continuous_action ({'tag': ['pr-331']})   |
|:--------------------------|:-------------------------------------------------------------|:----------------------------------------------|
| Ant-v4                    | 1808.30 ± 840.75                                             | -6.33 ± 2.75                                  |
| HalfCheetah-v4            | 2621.80 ± 1065.70                                            | 2149.96 ± 787.94                              |
| Hopper-v4                 | 2718.36 ± 306.63                                             | 1561.23 ± 202.97                              |
| InvertedDoublePendulum-v4 | 5703.07 ± 339.56                                             | 301.79 ± 15.95                                |
| Reacher-v4                | -4.65 ± 0.71                                                 | -66.57 ± 0.65                                 |
| Swimmer-v4                | 124.63 ± 21.98                                               | 117.50 ± 9.64                                 |
| Pusher-v4                 | -30.35 ± 6.34                                                | -277.30 ± 26.16                               |

Learning curves:
![mujoco_v4_part2_0_05](https://user-images.githubusercontent.com/15100273/209893934-34dd321a-ee24-4afc-ba42-4673ddf22286.png)

I have results for `alpha=0.05, 0.1` in the https://wandb.ai/openrlbenchmark/cleanrl for all these environments. Typically, results get better as we lower the value of `alpha`, and perform better (even better than PPO in some environments).

Reply:
Thanks for sharing these results. Some thoughts:

1. These results are fine as long as the settings to reproduce them are listed in `rpo.sh` and in the docs. 
2. You should also have a section in the docs that list out these "failed" experiments (probably via a [collapsable block](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks)), that way, the readers will be aware of this issue.
3. What does the performance look like for dm_control environments when `alpha=0.05`? It would be most desirable to find a single `alpha` that works strictly equal to or better than the default PPO in 95% of the environments tested.

Reply:
1 & 2: Sure, I will add them to the docs.

3. I do not have dm_control results with `alpha=0.05`. I have done an ablation study on a subset of them, and `alpha=0.5` seems to do better across these environments (Figure 17, https://arxiv.org/pdf/2212.07536.pdf). It seems to me that the choice of `alpha` depends on the environment's design (e.g., action scaling); thus, the optimal value might vary across environments (most likely across benchmarks).

Reply:
As it stands now: `alpha=0.5` is strictly equal to or better than the default PPO in 89% of environments tested (all 48/48 dm_control, 2/2 Gym, 7/11 mujoco_v4, 7/11 mujoco_v2. Total 64 out of 72 environments tested).

Reply:
> As it stands now: alpha=0.5 is strictly equal to or better than the default PPO in 89% of environments tested (all 48/48 dm_control, 2/2 Gym, 7/11 mujoco_v4, 7/11 mujoco_v2. Total 64 out of 72 environments tested).

This makes sense. Let's put this notice (RPO with alpha=0.5 is strictly equal to or better than the default PPO in 89% of environments ) in the PPO deprecation notice.

Reply:
Thoughts:
I do not see much difference between mujoco v4 and v2 in performance comparison. This is because they are the same environments. Is there any fundamental design difference between those environments? If not, then can we skip mujoco v2 environments for the comparison? It will make: RPO `alpha=0.5` strictly equal to or better than the default PPO in 93% of environments tested (all 48/48 dm_control, 2/2 Gym, 7/11 mujoco_v4. Total 57 out of 61 environments tested). Please let me know what you think. @vwxyzjn

Reply:
That's a good point. Then let's use this updated stats, but we probably should keep the v2 environment results for compatibility purposes :)

Reply:
Yes, sure.